main/src/library/MutDeque.flix

```diff
@@ -27,12 +27,23 @@
 /// Otherwise, the front index always points to an element (going counter-clockwise)
 /// and the back index always points to the first empty index (going clockwise).
 ///
-pub enum MutDeque[a] {
-    case MutDeque(Ref[Array[a]], Ref[Int32], Ref[Int32])
+pub enum MutDeque[a: Type, r: Region] {
+    case MutDeque(ScopedRef[ScopedArray[a, r], r], ScopedRef[Int32, r], ScopedRef[Int32, r])
 }
 
+
+instance Newable[MutDeque[a]] {
+    pub def new(r: Region[r]): MutDeque[a, r] \ Write(r) = MutDeque.new(r)
+}
+
+instance Scoped[MutDeque[a]] {
+    pub def regionOf(_: MutDeque[a, r]): Region[r] = () as Region[r]
+}
+
+// Find fix for this
```

Comment:
I think we can start by casting to impure.

Reply:
Right. We actually need to change the type of Iterable.

I suggest we just wait with the iterable instance, as you have been doing.

Reply:
Roger